### ISSUE TITLE
Ext_Authz and Ext_Proc Duration Custom Access Logs

### ISSUE THREAD

**Issue by llu94** [envoyproxy/envoy#19255]
*Title*: *Ext_Authz and Ext_Proc Duration Custom Access Logs*

*Description*:
I would like to add custom access logs that contain duration/timing information to Envoy's external authorization and external processing filters. 

[optional *Relevant Links*:]
This issue was inspired by this PR: https://github.com/envoyproxy/envoy/pull/18976

**Comment by phlax** [comment]
cc @alyssawilk

**Comment by llu94** [comment]
Added my PR

**Comment by esmet** [comment]
I mentioned this in the PR but I think this is a reasonable use case for dynamic metadata. We can have the ext_authz and ext_proc emit timing information as part of their dynamic metadata (possibly guarded by config) and then have users consume that dynamic metadata, which may end up being the access log or other filters.

**Comment by llu94** [comment]
> I mentioned this in the PR but I think this is a reasonable use case for dynamic metadata. We can have the ext_authz and ext_proc emit timing information as part of their dynamic metadata (possibly guarded by config) and then have users consume that dynamic metadata, which may end up being the access log or other filters.

I agree with this now that you've pointed it out and am willing to rework this. Do you have any examples for guarding dynamic metadata via configs just so I am clear what you are talking about?

**Comment by ggreenway** [comment]
Unless it will consume a non-trivial amount of CPU or memory, I think you can just add the dynamic metadata by default without config. My guess is that it will be far less expensive than the actual grpc request for ext_auth or ext_proc.

**Comment by llu94** [comment]
@ggreenway @phlax @esmet I reworked the feature in line with the dynamic metadata approach. I stumbled upon a comment in the ext_proc filter code indicating that dynamic metadata is not completely configured for ext_proc and a quick skin of the Well Known Dynamic Metadata docs indicates that this is probably true.

I'm willing to cut a PR to add dynamic metadata to the external processing filter after the new year and some discussion with interested stakeholders.
